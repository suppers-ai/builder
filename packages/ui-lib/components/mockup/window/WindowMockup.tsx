import { BaseComponentProps } from "../../types.ts";
import { ComponentChildren } from "preact";

// Window Mockup interfaces
export interface WindowMockupProps extends BaseComponentProps {
  /** Child content to display in the window */
  children: ComponentChildren;
  /** Window title */
  title?: string;
  /** Window variant */
  variant?: "default" | "bordered" | "shadow";
  /** Whether to show window controls */
  showControls?: boolean;
  /** Click handler for the mockup */
  onMockupClick?: () => void;
}

export function WindowMockup({
  children,
  title = "Window",
  variant = "default",
  showControls = true,
  class: className,
  ...props
}: WindowMockupProps) {
  // Build mockup classes
  const mockupClasses = [
    "mockup-window",
    variant === "bordered" && "border",
    variant === "shadow" && "shadow-2xl",
    "bg-base-300",
    className,
  ].filter(Boolean).join(" ");

  return (
    <div className={mockupClasses} {...props}>
      {showControls && (
        <div className="flex justify-between items-center px-4 py-2 bg-base-200 border-b">
          <div className="flex space-x-2">
            <div className="w-3 h-3 bg-error rounded-full"></div>
            <div className="w-3 h-3 bg-warning rounded-full"></div>
            <div className="w-3 h-3 bg-success rounded-full"></div>
          </div>
          <div className="text-sm font-medium">{title}</div>
          <div className="w-12"></div>
        </div>
      )}
      <div className="flex justify-center px-4 py-16 bg-base-200">
        {children}
      </div>
    </div>
  );
}
