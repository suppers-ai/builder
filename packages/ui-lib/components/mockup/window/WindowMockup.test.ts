import { assertEquals, assertStringIncludes } from "https://deno.land/std@0.224.0/assert/mod.ts";
import { assertSnapshot } from "https://deno.land/std@0.224.0/testing/snapshot.ts";
import { renderToString } from "preact-render-to-string";
import { WindowMockup } from "./WindowMockup.tsx";

Deno.test("WindowMockup - basic rendering", () => {
  const html = renderToString(WindowMockup({
    children: "Test content",
  }));
  assertStringIncludes(html, "mockup-window");
  assertStringIncludes(html, "Test content");
});

Deno.test("WindowMockup - with custom title", () => {
  const html = renderToString(WindowMockup({
    children: "Test content",
    title: "Custom Window Title",
  }));
  assertStringIncludes(html, "Custom Window Title");
});

Deno.test("WindowMockup - without controls", () => {
  const html = renderToString(WindowMockup({
    children: "Test content",
    showControls: false,
  }));
  assertStringIncludes(html, "mockup-window");
  // Should not include control buttons when controls are hidden
  assertEquals(html.includes("bg-red-500"), false);
  assertEquals(html.includes("bg-yellow-500"), false);
  assertEquals(html.includes("bg-green-500"), false);
});

Deno.test("WindowMockup - bordered variant", () => {
  const html = renderToString(WindowMockup({
    children: "Test content",
    variant: "bordered",
  }));
  assertStringIncludes(html, "border");
});

Deno.test("WindowMockup - shadow variant", () => {
  const html = renderToString(WindowMockup({
    children: "Test content",
    variant: "shadow",
  }));
  assertStringIncludes(html, "shadow-2xl");
});

Deno.test("WindowMockup - with window controls", () => {
  const html = renderToString(WindowMockup({
    children: "Test content",
    showControls: true,
  }));
  // Should include the colored control buttons
  assertStringIncludes(html, "bg-red-500");
  assertStringIncludes(html, "bg-yellow-500");
  assertStringIncludes(html, "bg-green-500");
});

Deno.test("WindowMockup - with custom class", () => {
  const html = renderToString(WindowMockup({
    children: "Test content",
    class: "custom-window-class",
  }));
  assertStringIncludes(html, "custom-window-class");
});

Deno.test("WindowMockup - HTML snapshot", async (t) => {
  const html = renderToString(WindowMockup({
    children: "Test content for snapshot",
    title: "Test Window",
  }));
  await assertSnapshot(t, html);
});

Deno.test("WindowMockup - HTML snapshot bordered variant", async (t) => {
  const html = renderToString(WindowMockup({
    children: "Bordered window content",
    variant: "bordered",
    title: "Bordered Window",
  }));
  await assertSnapshot(t, html);
});

Deno.test("WindowMockup - HTML snapshot without controls", async (t) => {
  const html = renderToString(WindowMockup({
    children: "No controls content",
    showControls: false,
    title: "Clean Window",
  }));
  await assertSnapshot(t, html);
});
