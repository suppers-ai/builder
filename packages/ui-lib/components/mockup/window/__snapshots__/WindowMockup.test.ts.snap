export const snapshot = {};

snapshot[`WindowMockupDisplay - HTML snapshot 1`] = `'<div class="mockup-window bg-base-300"><div class="flex justify-between items-center px-4 py-2 bg-base-200 border-b"><div class="flex space-x-2"><div class="w-3 h-3 bg-red-500 rounded-full"></div><div class="w-3 h-3 bg-yellow-500 rounded-full"></div><div class="w-3 h-3 bg-green-500 rounded-full"></div></div><div class="text-sm font-medium">Test Window</div><div class="w-12"></div></div><div class="flex justify-center px-4 py-16 bg-base-200">Test content for snapshot</div></div>'`;

snapshot[`WindowMockupDisplay - HTML snapshot bordered variant 1`] = `'<div class="mockup-window border bg-base-300"><div class="flex justify-between items-center px-4 py-2 bg-base-200 border-b"><div class="flex space-x-2"><div class="w-3 h-3 bg-red-500 rounded-full"></div><div class="w-3 h-3 bg-yellow-500 rounded-full"></div><div class="w-3 h-3 bg-green-500 rounded-full"></div></div><div class="text-sm font-medium">Bordered Window</div><div class="w-12"></div></div><div class="flex justify-center px-4 py-16 bg-base-200">Bordered window content</div></div>'`;

snapshot[`WindowMockupDisplay - HTML snapshot without controls 1`] = `'<div class="mockup-window bg-base-300"><div class="flex justify-center px-4 py-16 bg-base-200">No controls content</div></div>'`;
